src/main/java/org/beanpod/switchboard/entity/ChannelEntity.java

```diff
@@ -0,0 +1,28 @@
+package org.beanpod.switchboard.entity;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import lombok.*;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.ManyToMany;
+import java.util.Set;
+
+@Getter @Setter
+@AllArgsConstructor
+@NoArgsConstructor
+@Entity(name="Channel")
+@JsonIgnoreProperties({"hibernateLazyIntializer", "handler", "out", "in"})
+public class ChannelEntity {
+
+    @Id
+    Integer port;
```

Comment:
Ports are unique amongst themselves and not accross encoders and decoders

Reply:
not sure if i understand what you mean here.
multiple encoders can still have channels that use the same port.
if name is a problem it can be made as part of the primary key or removed.

Reply:
the issue was clarifiedd